qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -404,6 +404,15 @@ def simplify(self, atol=None, rtol=None):
             x = self.paulis.x[non_zero_indexes]
             z = self.paulis.z[non_zero_indexes]
             coeffs = coeffs[non_zero]
+
+            # Remove real (resp. imaginary) parts that are close to 0. This part for instance
+            # truncates 1+1e-17j to 1.0. Note that all coefficients that are left at this point
+            # are guaranteed to have a value above tolerance due to the previous is_zero check.
+            realpart_zero = np.isclose(coeffs.real, 0, atol=atol, rtol=rtol)
+            imagpart_zero = np.isclose(coeffs.imag, 0, atol=atol, rtol=rtol)
+            coeffs.real[realpart_zero] = 0
+            coeffs.imag[imagpart_zero] = 0
+
```

Comment:
I wouldn't do this; it's just encoding floating-point errors deliberately and slowing down the function.  The reason to have a tidy-up in a class such as this is to remove elements that have become zero overall, to reduce the number of elements actually stored in the object.  Removing the real or imaginary parts separately adds two more iterations through every child without removing anything or speeding things up; it just makes all floating-point errors larger when using `SparsePauliOp`.  There's no speed benefit for _part_ of a complex number becoming zero (maths on them still needs to handle the full complex number), and the current implementation may set some coefficients to zero, while leaving them in the object.

Reply:
I understand the concern and this change is not meant to speed up anything. But if we want to pass a `SparsePauliOp` into objects that don't allow imaginary parts (like the `PauliEvolutionGate`) we somehow need to be able to remove spurious complex parts. 

We could force users to do something like
```python
op = # some SparsePauliOp
op.coeffs = op.coeffs.real
```
but that seems a bit hacky and you can really only do that if we're sure there are no complex parts left...

Reply:
That's exactly what users _should_ do.  That operation is quite unusual, and the validity of it is very dependent on what the user is actually doing at any given time - it's not good to put it in an unrelated function.

The speed-up thing is relevant because that's what `SparsePauliOp.simplify` is meant to do; its contract is to sum duplicate representations of the same `Pauli` and remove ones whose coefficient becomes zero, for speed.  This PR changes that to be "chop small real and imaginary components", which is a floating-point error unless the user specifically asked for that, and it significantly slows down a core part of a sparse object by adding 2+ more iterations through all its coefficients.

If you really want this in `SparsePauliOp`, you should separate it out into another function (Mathematica calls this operation "chop", if you need a name), and remove the `rtol` parameter (`rtol` has no meaning when comparing to zero).  You should also ensure that it removes entries that become zero as a result of the chop.

Reply:
Or -- we could solve this differently altogether. We could leave `SparsePauliOp` as is and only add a `chop` argument to `Z2Symmetries` which chops real/imaginary parts after the tapering.

Reply:
In this particular situation, it's probably best to put it on `SparsePauliOp`, because there's a simplification step to it as well - if both the real and imaginary parts are chopped to 0, then you should remove that Pauli from the list.  That's functionality specific to `SparsePauliOp`, whereas the general `chop` would be at the Numpy level.